test/dok_matrix_test.rb

```diff
@@ -146,10 +146,50 @@ def test_multiply
     assert_equal(exp, actual.to_matrix, 'Matrix multiplication failed')
   end
 
+  def test_add_dok
+    test = SparseMatrixGenerator.generate_sparse_matrix(4, 4)
+    sparse_test = SparseMatrixFactory.build(test, DoKMatrixFactory.new)
+    actual = @sparse_matrix + sparse_test
+    exp = @matrix + test
+    assert_equal(exp, actual.to_matrix, 'Matrix addition failed')
+  end
+
+  def test_subtract_dok
+    test = SparseMatrixGenerator.generate_sparse_matrix(4, 4)
+    sparse_test = SparseMatrixFactory.build(test, DoKMatrixFactory.new)
+    actual = @sparse_matrix - sparse_test
+    exp = @matrix - test
+    assert_equal(exp, actual.to_matrix, 'Matrix subtraction failed')
+  end
+
+  def test_multiply_dok
+    test = SparseMatrixGenerator.generate_sparse_matrix(4, 4)
+    sparse_test = SparseMatrixFactory.build(test, DoKMatrixFactory.new)
+    actual = @sparse_matrix * sparse_test
+    exp = @matrix * test
+    assert_equal(exp, actual.to_matrix, 'Matrix multiplication failed')
+  end
+
   def test_power
-    scalar = rand(-10..10)
-    actual = @sparse_matrix**scalar
-    exp = @matrix**scalar
-    assert_equal(exp, actual.to_matrix, 'Matrix exponentiation failed')
+    scalar = rand(0..10)
```

Comment:
I would make two tests one for a regular and one for a irregular matrix to validate matrix power.
(These don't necessarily have to be based of randomly generated like this test).

Reply:
Better not to use rand(0..10) anyways as generating 1 will be a trivial case and always pass.

Reply:
I used rand (2..5) in my own PR for csr_matrix

Reply:
I can also fix dis